### PR TITLE
CLI - Integrage result summary and update output format of data diagnosis

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
Integrage result summary and update output format of data diagnosis.

**Major Revision**
- integrage result summary 
- add md and html format for data diagnosis

### CODE REVIEW

File: docs/cli.md [hunk 1]
```diff
@@ -188,7 +189,8 @@ sb result diagnosis [--baseline-file]
 | Name                   | Default | Description                                                                 |
 |------------------------|---------|-----------------------------------------------------------------------------|
 | `--output-dir`         | `None`  | Path to output directory, outputs/{datetime} will be used if not specified. |
-| `--output-file-format` | `excel` | Format of output file, excel or json. Default: excel.                       |
+| `--output-file-format` | `excel` | Format of output file, 'excel', 'json', 'md' or 'html'. Default: excel.                       |
```
Comment (cp5555) [inline docs/cli.md:192]: do we need to align with `|`

File: docs/cli.md [hunk 2]
```diff
@@ -208,6 +210,56 @@ Run data diagnosis and output the results in jsonl format:
...
+| `--data-file` `-d`     | Path to raw data file. |
+| `--rule-file` `-r`     | Path to rule file.     |
+
+#### Optional arguments
+
+| Name                   | Default | Description                                                                 |
+|------------------------|---------|-----------------------------------------------------------------------------|
+| `--output-dir`         | `None`  | Path to output directory, outputs/{datetime} will be used if not specified. |
+| `--output-file-format` | `md` | Format of output file, 'excel', 'md' or 'html'. Default: md.                       |
+| `--round` | 2 | Number of valid decimal places to show in output. Default: 2.                       |
```
Comment (cp5555) [inline docs/cli.md:243]: aligment as well. `Number of valid decimal places to show in output. Default: 2. `, what's the meaning of `valid decimal places`?

File: docs/cli.md [hunk 3]
```diff
@@ -173,6 +173,7 @@ sb result diagnosis [--baseline-file]
        [--rule-file]
        [--output-dir]
        [--output-file-format]
+       [--round]
```
Comment (cp5555) [inline docs/cli.md:176]: decimal-place-value

File: docs/cli.md (hunk 2 above)
Comment (yukirora) [reply docs/cli.md]: ok, updated

File: docs/cli.md [hunk 4]
```diff
@@ -173,6 +173,7 @@ sb result diagnosis [--baseline-file]
        [--rule-file]
        [--output-dir]
        [--output-file-format]
+       [--decimal-place-value]
```
Comment (abuccts) [inline docs/cli.md:176]: sort in alphabet order?

File: docs/cli.md [hunk 5]
```diff
@@ -185,10 +186,11 @@ sb result diagnosis [--baseline-file]
...
 
-| Name                   | Default | Description                                                                 |
-|------------------------|---------|-----------------------------------------------------------------------------|
-| `--output-dir`         | `None`  | Path to output directory, outputs/{datetime} will be used if not specified. |
-| `--output-file-format` | `excel` | Format of output file, excel or json. Default: excel.                       |
+| Name                    | Default | Description                                                                 |
+|-------------------------|---------|-----------------------------------------------------------------------------|
+| `--output-dir`          | `None`  | Path to output directory, outputs/{datetime} will be used if not specified. |
+| `--output-file-format`  | `excel` | Format of output file, 'excel', 'json', 'md' or 'html'. Default: excel.     |
+| `--decimal-place-value` | 2       | Number of valid decimal places to show in output. Default: 2.               |
```
Comment (abuccts) [inline docs/cli.md:193]: same, move to the beginning?

File: docs/cli.md [hunk 6]
```diff
@@ -208,6 +210,56 @@ Run data diagnosis and output the results in jsonl format:
 sb result diagnosis --data-file outputs/results-summary.jsonl --rule-file rule.yaml --baseline-file baseline.json --output-file-format json
 ```
 
+Run data diagnosis and output the results in markdown format with 2 valid decimal places:
+```bash title="SB CLI"
+sb result diagnosis --data-file outputs/results-summary.jsonl --rule-file rule.yaml --baseline-file baseline.json --output-file-format md --decimal-place-value 2
+```
+
+### `sb result summary`
```
Comment (abuccts) [inline docs/cli.md:218]: do you need to add a tutorial in document like [data diagnosis](https://microsoft.github.io/superbenchmark/docs/user-tutorial/data-diagnosis)?

**Review by abuccts — APPROVED** [review]

File: docs/cli.md (hunk 4 above)
Comment (yukirora) [reply docs/cli.md]: updated, thanks

File: docs/cli.md (hunk 6 above)
Comment (yukirora) [reply docs/cli.md:218]: sure, I plan to submit another pr for that

File: docs/cli.md (hunk 5 above)
Comment (yukirora) [reply docs/cli.md]: updated

File: docs/cli.md [hunk 7]
```diff
@@ -170,6 +170,7 @@ Filter the defective machines automatically from benchmarking results according
 ```bash title="SB CLI"
 sb result diagnosis [--baseline-file]
        [--data-file]
+       [--decimal-place-value]
        [--rule-file]
        [--output-dir]
        [--output-file-format]
```
Comment (abuccts) [inline docs/cli.md:176]:
```suggestion
sb result diagnosis --baseline-file
                    --data-file
                    --rule-file
                    [--decimal-place-value]
                    [--rule-file]
                    [--output-dir]
                    [--output-file-format {excel, json}]
```
pls follow the [format here](https://docs.microsoft.com/en-us/cli/azure/vm?view=azure-cli-latest#az-vm-create)

File: docs/cli.md [hunk 8]
```diff
@@ -208,6 +210,56 @@ Run data diagnosis and output the results in jsonl format:
...
+### `sb result summary`
+
+Generate the readable summary report automatically from benchmarking results according to rules defined in rule file.
+
+```bash title="SB CLI"
+sb result summary [--data-file]
+       [--decimal-place-value]
+       [--rule-file]
+       [--output-dir]
+       [--output-file-format]
```
Comment (abuccts) [inline docs/cli.md:227]:
```suggestion
sb result summary --data-file
                  --rule-file
                  [--decimal-place-value]
                  [--output-dir]
                  [--output-file-format {md, excel, html}]
```

File: docs/cli.md (hunk 7 above)
Comment (yukirora) [reply docs/cli.md]: updated, thanks

File: docs/cli.md (hunk 8 above)
Comment (yukirora) [reply docs/cli.md]: updated

**Review by abuccts — APPROVED** [review]

**Pull request merged**